File: DSA/NeeravKumarSharma_590018263/Day_12/Q1_LeetCode.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int lines = 1;
        int currentWid = 0;
        
        for (char c : s.toCharArray()) {
            int charWidth = widths[c - 'a'];
            
            if (currentWid + charWidth > 100) {
                lines++;
                currentWid = charWidth;
            } else {
                currentWid += charWidth;
            }
        }
        
        return new int[]{lines, currentWid};
    }
}

File: DSA/Aarush_590017879/Day_12/Q1_leetcode.c

int* numberOfLines(int* widths, int widthsSize, char* s, int* returnSize) {
    int lines = 1;
    int currentWidth = 0;

    for (int i = 0; s[i] != '\0'; i++) {
        int w = widths[s[i] - 'a'];

        if (currentWidth + w > 100) {
            lines++;
            currentWidth = w;
        } else {
            currentWidth += w;
        }
    }

    int* result = (int*)malloc(2 * sizeof(int));
    result[0] = lines;
    result[1] = currentWidth;

    *returnSize = 2;
    return result;
}